File: IncomeTaxCalculator.java
import java.util.Scanner;

public class IncomeTaxCalculator {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);

        // Take input from user and displays net salary
        System.out.print("Enter your gross monthly salary (GH¢): ");
        double grossSalary = input.nextDouble();
        double answer = IncomeTax(grossSalary);
        System.out.printf("Net Income: GH¢%.2f%n", answer);
        input.close();
    }

    public static double IncomeTax(double grossIncome) {
        double tax = 0;
        double remainingIncome = grossIncome;

        if (remainingIncome <= 402) {
        } else if (remainingIncome <= 512) {
            tax += (remainingIncome - 402) * 0.055;
        } else if (remainingIncome <= 642) {
            tax += (remainingIncome - 512) * 0.1 + 5.5;
        } else if (remainingIncome <= 3642) {
            tax += (remainingIncome - 642) * 0.175 + 18.5;
        } else if (remainingIncome <= 20037) {
            tax += (remainingIncome - 3642) * 0.25 + 543.5;
        } else if (remainingIncome <= 50000) {
            tax += (remainingIncome - 20037) * 0.30 + 4642.25;
        } else if (remainingIncome > 50000) {
            tax += (remainingIncome - 50000) * 0.35 + 13631.15;
        }

        double netSalary = grossIncome - tax;
        return netSalary;
    }
}

File: JavaProjects/income_tax_calculator.py
def income_tax(gross_income: float) -> float:
    cumulative_incomes = [402, 512, 642, 3642, 20037, 50000]
    cumulative_tax = [0, 5.5, 18.5, 543.5, 4642.25, 13631.15]
    tax_rates = [0, 5, 10, 17.5, 25, 30, 35]

    tax = 0
    remaining_income = gross_income

    for i in range(len(cumulative_incomes)):
        if remaining_income <= cumulative_incomes[i]:
            tax += cumulative_tax[i - 1]
            remaining_income -= cumulative_incomes[i - 1]
            tax += remaining_income * (tax_rates[i] / 100)
            break
        else:
            tax += cumulative_tax[-1]
            remaining_income -= cumulative_incomes[-1]
            tax += remaining_income * (tax_rates[-1] / 100)
            break

    net_salary = gross_income - tax
    return net_salary


# Takes input from the user
gross_salary = float(input("Enter your gross monthly salary (GH¢): "))
if "__main__" == __name__:
    # Calculates tax and net salary
    net_salary = income_tax(gross_salary)

# Displays results
print(f"Net Monthly Salary: GH¢{net_salary:.2f}")
